app/Gists/GistConfig.php

```diff
@@ -0,0 +1,83 @@
+<?php namespace Gistlog\Gists;
+
+use Illuminate\Support\Arr;
+use Symfony\Component\Yaml\Yaml;
+
+class GistConfig implements \ArrayAccess
+{
+
+    /**
+     * @var array
+     */
+    private $default_settings = array(
+        'published'    => false,
+        'published_on' => null,
+        'preview'      => null
+    );
+
+    /**
+     * @var array
+     */
+    public $settings = array();
+
+    /**
+     * @param array|ArrayAccess $githubGist
+     * @return GistConfig
+     */
+    public static function fromGitHub($githubGist)
+    {
+        $config = new self;
+        $config->settings = $config->default_settings;
+
+        if (! array_key_exists('gistlog.yml', $githubGist['files'])) {
+            return $config;
+        }
+
+        $user_settings = Yaml::parse($githubGist['files']['gistlog.yml']['content']);
```

Comment:
Did some checking on this over at https://developer.github.com/v3/gists and I didn't see any cases in which that would be a concern. Should always contain the `content` key